### PR TITLE
Timeline, Funnel, and Tree Map Panels

### PR DESCRIPTION
**Pull request by timio23**

### CODE REVIEW

**Review by HimoriTempest — CHANGES_REQUESTED** [review]
Greetings @timio23 ,
my tests revealed that your extensions compile but don't load.
Would you be able to share under which version this has been tested on your end?
Keep in mind that all extensions should be working on version 11.0.0 and higher.

**Comment by timio23** [comment]
Hi @HimoriTempest, that's odd. I test on versions 11.1.1 and 10.13.3. Did you get any console errors?

**Comment by timio23** [comment]
@HimoriTempest, found a missing dependancy on the funnel chart. Does that fix the issue?

**Review by HimoriTempest — CHANGES_REQUESTED** [review]
Looks good to me 🚀 
I don't see the Y and X axis labels.
Are they hidden on purpose?

File: packages/funnel-chart-panel/src/panel.vue [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+		<VProgressCircular v-else-if="isLoading" :indeterminate="true"/>
+		<v-info type="danger" icon="error" :center="true" v-else-if="!isLoading && !chartEl" title="Incompatible Data">The current data is not compatiple with the scatter plot.</v-info>
+		<div ref="chartEl" />
+	</div>
+</template>
+
+<script lang="ts">
+import { useI18n } from 'vue-i18n';
+import { useApi, useStores } from '@directus/extensions-sdk';
+import { abbreviateNumber } from '@directus/shared/utils';
```
[the remark this answers is not in the document]
Comment (timio23) [reply packages/funnel-chart-panel/src/panel.vue:16]: I'll look into the total time always displaying, but as for the X and Y labels, these are hidden when the panel is small. I decided to follow suite of the built-in Directus charts which do the same thing.

**Review by HimoriTempest — COMMENTED** [review]
Looks good to me🚀

File: packages/funnel-chart-panel/src/panel.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (HimoriTempest) [reply packages/funnel-chart-panel/src/panel.vue:16]: Thanks for the clarification

**Review by HimoriTempest — APPROVED** [review]

**Comment by phazonoverload** [comment]
Hi! Small final changes as a result of our creative review.

## Funnel Chart
- Text in tooltip does not match to the funnel color change
- Tooltip background needs to change from white to dark according to the modes.
- The 'gaps' between the segments - can they also be darker in dark mode or is it a pain? 

## Timeline Chart
- Text in tooltip does not match to the timeline color change
- Start Date + Date End field should we together side-by-side in options - can you put the two sort options and the two date options together? 
- The bar border radius seems to be 0px compared to the other charts with bars that have a slightly rounded corners.
- Tooltip background needs to change from white to dark according to the modes.
- All the colors have an partial opacity of 0.66 - can it be at 1 please? 
- remove option to do multi-series (grouping). 
- Add a delimiter between the task name and number of days (suggesting a colon)
- In dark mode, remove the 

## Tree Map Chart
- Change the white border to fit the dark mode theme
- Remove all hover/active states for segments.
- Can we lower the range between the base color and the lightest color? Lightest ones can be too light. 
- Remove the hashes/lines on the very bottom. Can be quite noticable in dark mode.
  image
- Can we show the value on the data label after a colon. Currently “Phase 2” should it be “Phase 2: 000"

**Comment by timio23** [comment]
@BB-Loft I've uploaded the latest changes.

@phazonoverload, thanks for your suggestions. I have the following notes:
- Apexcharts doesn't have an option to change the funnel chart gap color. I can change it using CSS but it wouldn't match the chosen colour anymore. It would be need to be a static value or a css theme variable.
- The timeline opacity is for overlapping timelines. Do you still want the opacity changed to 1?
image
- I've removed the grouping from the timeline chart but since it's optional, wouldn't it be better to leave it in?
- Rather than remove the hover tooltip on the treetop charts, I've disabled the tooltip by default. If they choose to hide the labels, I think it is worth having the option for tooltips. What are your thoughts?

All other changes have been made.

**Comment by phazonoverload** [comment]
> Apexcharts doesn't have an option to change the funnel chart gap color. I can change it using CSS but it wouldn't match the chosen colour anymore. It would be need to be a static value or a css theme variable. 

Happy to leave it as-is given this. 

> The timeline opacity is for overlapping timelines. Do you still want the opacity changed to 1?

Also no - makes total sense. 

> I've removed the grouping from the timeline chart but since it's optional, wouldn't it be better to leave it in?

Nah - I'm a fan of more simple extensions at this point. Also, there was a significant creative review feedback for this mode that would have pushed our release deadline. 

> Rather than remove the hover tooltip on the treetop charts, I've disabled the tooltip by default. If they choose to hide the labels, I think it is worth having the option for tooltips. What are your thoughts?

Agree! 

@HimoriTempest can you give this one last look-over, but I think it might be good to go.

**Comment by phazonoverload** [comment]
Oh just a small note @timio23 we'll need to redo the graphic in the readme for the timeline to not show grouping

**Comment by timio23** [comment]
@phazonoverload, I've updated that screenshot now

**Comment by HimoriTempest** [comment]
@phazonoverload Looking good 🚀

**Pull request merged**